db/mongoTest.js

```diff
@@ -0,0 +1,93 @@
+const mongoose = require('mongoose');
+const Projects = require('./Project.js').project;
+const Users = require('./Project.js').user;
+
+async function getProjAndBackers(id) {
+  // Commented-out lines used when invoking directly, not as helper
+  // await mongoose.connect('mongodb://127.0.0.1/quickstarter');
+  try {
+    const start = new Date().getTime();
+    const project = await Projects.findById(id);
+    const backers = await Users.find({ _id: { $in: project.backerIds } });
+    const end = new Date().getTime();
+    const time = end - start;
+    // console.log(`Milliseconds: ${time}`);
+    // console.log(`Number of backers: ${backers.length}`);
+    // mongoose.disconnect();
+    return time;
+  } catch (error) {
+    console.error(error);
+    mongoose.disconnect();
+  }
+}
+
+async function testManyRecordsParallel(n) {
+  const promises = [];
+  let countdown = n;
+  await mongoose.connect('mongodb://127.0.0.1/quickstarter');
+  const start = new Date().getTime();
+  while (countdown--) {
+    const id = Math.floor(Math.random() * 10000000);
+    promises.push(getProjAndBackers(id));
+  }
+  const times = await Promise.all(promises);
+  const end = new Date().getTime();
+  mongoose.disconnect();
+  const averageRequestTime = times.reduce((memo, time) => memo + time) / n;
```

Comment:
What was the reason you terminated mongo connection and restarted it every time?

Reply:
These are outside the loop where queries are actually performed, so I only open the connection before sending all the queries and close it after they're all done (with the `await Promise.all(promises)`).